cmd/watch.go

```diff
@@ -65,12 +70,16 @@ var watchCmd = &cobra.Command{
 		}
 		watchPath := util.ReadFilePath(u, runtime.GOOS)
 
-		err = component.WatchAndPush(client, componentName, applicationName, watchPath, stdout)
+		err = component.WatchAndPush(client, componentName, applicationName, watchPath, stdout, ignores, delay)
 		checkError(err, "Error while trying to watch %s", watchPath)
 	},
 }
 
 func init() {
+	// ignore git as it can change even if no source file changed
+	// for example some plugins providing git info in PS1 doing that
+	watchCmd.Flags().StringSliceVar(&ignores, "ignores", []string{".*\\.git.*"}, "Files and/or folders to be recursively ignored for watch can be specified as regular expressions with this flag.")
+	watchCmd.Flags().IntVar(&delay, "delay", 1, "Time in seconds between a detection of code change and push.delay=0 means changes will be pushed as soon as they are detected which can cause performance issues")
```

Comment:
`Time in seconds between detection of code change and push` May be better as a shorter description. You also already specify an example, so no need to add `push.delay=0`

Reply:
@kadel suggested https://github.com/redhat-developer/odo/pull/742#discussion_r220519154

What do you say?

Reply:
I don't mind going with @kadel 's suggestion :+1: